multiqc/modules/sambamba/markdup.py

```diff
@@ -63,27 +64,91 @@ def get_markdup_stats(self, f):
         """
         Input actual content of sambamba markdup log output.
         Regex for important phrases and extract reads.
-        Calculate and return duplicate rate.
-        Outputs dictionary of markdup stats of a sample.
+        Detect and calculate duplicate rate by single/paired-end samples.
+        Outputs dictionary of markdup stats of a sample by single/paired-end type.
         This dict is associated with corresponding sample in key of markdup_data_dict.
         """
 
+        """NOTE: reads sum to between 98 - 99% of input BAM file. Is Sambamba Markdup not recognizing the entire file?"""
+
         sorted_end_pairs = int(re.search("sorted \d+ end pairs", f).group(0).split(" ")[1])
         single_ends = int(re.search("and \d+ single ends", f).group(0).split(" ")[1])
         single_unmatched_pairs = int(re.search("among them \d+ unmatched", f).group(0).split(" ")[2])
         duplicate_reads = int(re.search("found \d+ duplicates", f).group(0).split(" ")[1])
 
-        duplicate_rate = duplicate_reads / (sorted_end_pairs * 2 + single_ends - single_unmatched_pairs) * 100
-        duplicate_rate = round(duplicate_rate, 2)
+        duplicate_rate = self.calculate_duplicate_rate(
+            sortedEndPairs=sorted_end_pairs,
+            singleEnds=single_ends,
+            singleUnmatchedPairs=single_unmatched_pairs,
+            duplicateReads=duplicate_reads,
+        )
 
-        stats = {
-            "Sorted End Pairs": sorted_end_pairs,
-            "Single Ends": single_ends,
-            "Single Unmatched Pairs": single_unmatched_pairs,
-            "Duplicate Reads": duplicate_reads,
-            "Duplicate Rates": duplicate_rate,
-        }
+        sample_statistics = self.calculate_read_proportions(
+            sortedEndPairs=sorted_end_pairs,
+            singleEnds=single_ends,
+            singleUnmatchedPairs=single_unmatched_pairs,
+            duplicateReads=duplicate_reads,
+            duplicateRate=duplicate_rate,
+        )
+
+        return sample_statistics
+
+    def calculate_duplicate_rate(self, sortedEndPairs, singleEnds, singleUnmatchedPairs, duplicateReads):
+
+        """
+        Input all regexed stats from markdup log.
+        Detect if sample is paired or single-end.
+        Calculate duplicate rate as needed.
+        Print debug msg if a sample neither fits singleEnds or PE scheme.
+        Output duplicate rate.
+        """
+
+        if (sortedEndPairs > 0) and (singleUnmatchedPairs > 0):
+            duplicate_rate = duplicateReads / (sortedEndPairs * 2 + singleEnds - singleUnmatchedPairs) * 100
+            duplicate_rate = round(duplicate_rate, 2)
+        elif (sortedEndPairs == 0) and (singleUnmatchedPairs == 0):
+            duplicate_rate = duplicateReads / singleEnds * 100
+            duplicate_rate = round(duplicate_rate, 2)

```

Comment:
This code is the same as the above if `sortedEndPairs` and `singleUnmatchedPairs` are both `0` (`singleEnds - 0` is `singleEnds`, `sortedEndPairs * 2` is `0` if `sortedEndPairs` is `0`).

I think we can get rid of this `if` block..